ckan/migration/migrate_package_activity.py

```diff
@@ -67,15 +66,25 @@ def num_activities_migratable():
     from ckan import model
     num_activities = model.Session.execute(u'''
     SELECT count(*) FROM activity a JOIN package p ON a.object_id=p.id
-    WHERE a.activity_type IN ('new package', 'changed package')
-    AND p.private = false;
+    WHERE a.activity_type IN ('new package', 'changed package');
     ''').fetchall()[0][0]
     return num_activities
 
 
 def migrate_all_datasets():
     import ckan.logic as logic
-    dataset_names = logic.get_action(u'package_list')(get_context(), {})
+    dataset_names = []
+    offset = 0
+    count = 1
+    while len(dataset_names) < count:
+        dn = logic.get_action(u'package_search')(get_context(),
+                                                {'include_private': True,
+                                                'fl': ['id'],
+                                                'rows': 5000,
+                                                'start': offset})
+        offset += 5000
```

Comment:
```suggestion
        offset += len(dn['results'])
```
is safer in case an implementation sets a lower limit on records returned for each call